### ISSUE TITLE
Compilation fails with latest version of tsc

### ISSUE THREAD

**Issue by m1cm1c** [lukeautry/tsoa#1316]
Compilation works with ts-node but fails with the latest version of tsc.

## Sorting

- **I'm submitting a ...**

  - [x] bug report
  - [ ] feature request
  - [ ] support request

- I confirm that I
  - [x] used the [search](https://github.com/lukeautry/tsoa/search?type=Issues) to make sure that a similar issue hasn't already been submit

## Expected Behavior

Compile.

## Current Behavior

Does not compile:

```
node_modules/@tsoa/runtime/dist/decorators/middlewares.d.ts:7:65 - error TS2344: Type 'T' does not satisfy the constraint 'Object | Function'.
  Type 'T' is not assignable to type 'Function'.

7 export declare function Middlewares<T>(...mws: Array<Middleware<T>>): ClassDecorator & MethodDecorator;
                                                                  ~

  node_modules/@tsoa/runtime/dist/decorators/middlewares.d.ts:7:37
    7 export declare function Middlewares<T>(...mws: Array<Middleware<T>>): ClassDecorator & MethodDecorator;
                                          ~
    This type parameter might need an `extends Function` constraint.
  node_modules/@tsoa/runtime/dist/decorators/middlewares.d.ts:7:37
    7 export declare function Middlewares<T>(...mws: Array<Middleware<T>>): ClassDecorator & MethodDecorator;
                                          ~
    This type parameter might need an `extends Object | Function` constraint.

node_modules/@tsoa/runtime/dist/decorators/middlewares.d.ts:14:76 - error TS2344: Type 'T' does not satisfy the constraint 'Object | Function'.
  Type 'T' is not assignable to type 'Function'.

14 export declare function fetchMiddlewares<T>(target: any): Array<Middleware<T>>;
                                                                              ~

  node_modules/@tsoa/runtime/dist/decorators/middlewares.d.ts:14:42
    14 export declare function fetchMiddlewares<T>(target: any): Array<Middleware<T>>;
                                                ~
    This type parameter might need an `extends Function` constraint.
  node_modules/@tsoa/runtime/dist/decorators/middlewares.d.ts:14:42
    14 export declare function fetchMiddlewares<T>(target: any): Array<Middleware<T>>;
                                                ~
    This type parameter might need an `extends Object | Function` constraint.
```

## Steps to Reproduce

1. Install the latest version of TypeScript (the problem did not exist until recently, so make sure that `tsc --version` prints `Version 4.8.3`; Typescript's type system recently got a bit stricter.).
2. Drop [this package.json](https://gist.github.com/m1cm1c/96c1d7741f6599a78a5a85b38c60c5e7) and [this tsconfig.json](https://gist.github.com/m1cm1c/e4ceb54b18af46d1964ec5052ed3b08d) (the tsconfig doesn't really matter but there needs to be some) into an otherwise empty directory and run `yarn install`.
3. Create a directory `src` and in it a file `index.ts` containing `import {Controller, Request, Post, Body, Route, Tags, SuccessResponse, Response, Path, Security, Get, Query} from 'tsoa';`
4. Run `tsc` in the directory containing the tsconfig.

## Context (Environment)

Version of the library: 4.1.2
Version of NodeJS: 16.17.0

- Confirm you were using yarn not npm: [x] Exact same behavior with both npm and yarn.

**Comment by WoH** [comment]
Please confirm this is fixed in 4.1.3.

**Comment by m1cm1c** [comment]
@WoH Nice, thank you for fixing it this quickly! The problem is fixed in 4.1.3.
